src/WidgetRegistry.ts

```diff
@@ -113,7 +121,7 @@ export class WidgetRegistry extends Evented implements WidgetRegistry {
 		promise.then((widgetCtor) => {
 			this.registry.set(widgetLabel, widgetCtor);
 			this.emit({
-				type: `loaded:${widgetLabel}`
+				type: getEventType(widgetLabel)
```

Comment:
Sort of a macro thing... you are emitting the same event in 3 locations in code.  That seems like there is some duplicate logic here.